src/init.cpp

```diff
@@ -511,11 +512,21 @@ bool AppInit2(ThreadHandlerPtr threads)
 
     if (mapArgs.count("-timeout"))
     {
-        int nNewTimeout = GetArg("-timeout", 4000);
+        int nNewTimeout = GetArg("-timeout", 5000);
         if (nNewTimeout > 0 && nNewTimeout < 600000)
             nConnectTimeout = nNewTimeout;
     }
 
+    if (mapArgs.count("-peertimeout"))
+    {
+        int nNewPeerTimeout = GetArg("-timeout", 45);
```

Comment:
Should `-timeout` not be `-peertimeout`?

Reply:
no timeout is a socket timeout (when trying to connect to a address).and peer timeout is the non response from a open connected socket. i followed bitcoins area for this.

Reply:
-timeout is already an argument for socket timeout

Reply:
Yes, but in this case, it looks like we meant to `GetArg("-peertimeout", 45)` instead of `GetArg("-timeout", 45)`. The value of `nNewPeerTimeout` is set from the wrong program option, right?